middleware/launch.test.js

```diff
@@ -0,0 +1,49 @@
+const getLaunchUrl = require("./launch");
+const mockRepository = require("../tests/utils/mockRepository");
+
+let repositories;
+const QUESTIONNAIRE_ID = 1;
+
+describe("status middleware", () => {
+  let res;
+  let req = { params: { questionnaireId: QUESTIONNAIRE_ID } };
+  res = {
+    redirect: jest.fn()
+  };
+  let next = jest.fn();
+
+  repositories = {
+    Metadata: mockRepository({
+      findAll: [{ id: 1, key: "hello", type: "Text", value: "hello" }]
+    })
+  };
+
+  it("should find all metadata when the endpoint is hit", async () => {
+    const ctx = { repositories };
+    await getLaunchUrl(ctx)(req, res, next);
+
+    expect(repositories.Metadata.findAll).toHaveBeenCalledWith({
+      questionnaireId: QUESTIONNAIRE_ID
+    });
+  });
+
+  it("should call a redirect with a jwt", async () => {
+    const ctx = { repositories };
+    await getLaunchUrl(ctx)(req, res, next);
+
+    expect(res.redirect).toHaveBeenCalled();
+  });
+
+  it("should call next with an error if a metadata is missing a key", async () => {
```

Comment:
Probably also worth adding tests describing the behaviour of `null` and "    ". Since keys are null by default?